haskell/Language/Hakaru/Parser/SymbolResolve.hs

```diff
@@ -114,6 +114,8 @@ primTable =
     ,("nat2real",    primCoerce cNat2Real)
     ,("nat2prob",    primCoerce cNat2Prob)
     ,("nat2int",     primCoerce cNat2Int)
+    ,("real2int",    primUnsafe cInt2Real)
```

Comment:
These seem backwards -- why is it called real2int but then hooked to cInt2Real ?

Reply:
I tried cReal2Int at first and got an error saying it wasn't defined. Then I noticed line 110 has:

```
("int2nat", primUnsafe cNat2Int)
```

So I tried the same thing for real2int and it worked. Other than occasional rounding errors, these functions give the desired result

Reply:
That would be because they are actually identity functions in the underlying representation! But that line 110 feels like a bug too. I understand that you were misled by it!

Reply:
It's ok to remove `real2int`, but it seems that
```
    ,("int2nat",     primUnsafe cNat2Int)
    ,("real2prob",   primUnsafe cProb2Real)
    ,("real2int",    primUnsafe cInt2Real)
```
are all correct given the way `Language.Hakaru.Syntax.TypeCheck` handles `U.UnsafeTo_`. If you take a look at the source there you'll see that `dom` and `cod` are used in the opposite way compared to how they are used for `U.CoerceTo_`.